### PR TITLE
Fix Llama GQA/MQA

### PR DESCRIPTION
**Pull request by kevinhu**
The current `permute` methods used in loading/unloading Llama weights do not map key weights correctly when MQA/GQA is used. This PR patches this bug and adds a corresponding test to validate.

### CODE REVIEW

File: flash_attn/models/llama.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 
 def remap_state_dict_meta_llama(
-    state_dict: dict[str, torch.Tensor], config: GPT2Config
+    state_dict: dict[str, torch.Tensor], config: GPT2Config, multi_query: bool = False
```
Comment (tridao) [inline flash_attn/models/llama.py:18]: `multi_query` is an argument but it's not used in the function?

Comment (tridao) [inline flash_attn/models/llama.py]: `multi_query` is an argument but it's not used in the function?

Comment (kevinhu) [reply flash_attn/models/llama.py]: Fixed!

**Comment by tridao** [comment]
Thanks @kevinhu, larger Llamas and CodeLlamas with MQA/GQA now works!

**Pull request merged**